pages/_document.js

```diff
@@ -1,10 +1,27 @@
 import Document, { Head, Main, NextScript } from "next/document";
 
+import { gaTrackingId } from "constants/env";
+
 export default class MyDocument extends Document {
   render() {
     return (
       <html>
         <Head>
+          {/* Global Site Tag (gtag.js) - Google Analytics */}
+          <script
+            async
+            src={`https://www.googletagmanager.com/gtag/js?id=${gaTrackingId}`}
+          />
+          <script
+            dangerouslySetInnerHTML={{
+              __html: `
+            window.dataLayer = window.dataLayer || [];
+            function gtag(){dataLayer.push(arguments);}
+            gtag('js', new Date());
+            gtag('config', '${gaTrackingId}');
```

Comment:
I don't recall `Document` or `MyDocument` is used so just to sanity check, can you clarify when exactly this `gtag('config', '${gaTrackingId}')` pageview tracking event would be fired?  Is it just when `MainLayout` is rendered?

Reply:
this is the wrapper of any/every page with or without `MainLayout`. this should instantiate `gtag` on any server-side page render. client-side calls would not re-instantiate.